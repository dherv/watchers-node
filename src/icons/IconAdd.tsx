import { FC } from "react";
import styled from "styled-components";

const IconAdd: FC<{ onClick: (event: any) => void }> = ({ onClick }) => (
  <Button onClick={onClick}>
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width="24"
      height="24"
      viewBox="0 0 24 24"
      fill="#fff"
    >
      <title>Add</title>
      <path d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z" />
      <path d="M0 0h24v24H0z" fill="none" />
    </svg>
  </Button>
);
const Button = styled.button`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 32px;
  height: 32px;
  line-height: 32px;
  background-color: #484848;
  border: none;
  border-radius: 50%;
  cursor: pointer;
  &:hover {
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  }
`;
export default IconAdd;
