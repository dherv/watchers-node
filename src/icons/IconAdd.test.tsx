import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import "jest-styled-components";

import IconAdd from "./IconAdd";

describe("IconAdd", () => {
  const props = { onClick: jest.fn() };
  const wrapper = shallow(<IconAdd {...props} />);

  describe("IconAdd component", () => {
    describe("user interface", () => {
      test("", () => {});
    });
    describe("events", () => {});
  });

  describe("snapshot", () => {
    test("should match", () => {
      expect.assertions(1);
      const tree = renderer.create(<IconAdd {...props} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
