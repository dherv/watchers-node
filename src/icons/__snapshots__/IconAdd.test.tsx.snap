// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconAdd snapshot should match 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 32px;
  height: 32px;
  line-height: 32px;
  background-color: #484848;
  border: none;
  border-radius: 50%;
  cursor: pointer;
}

.c0:hover {
  box-shadow: 0px 4px 4px rgba(0,0,0,0.25);
}

<button
  className="c0"
  onClick={[MockFunction]}
>
  <svg
    fill="#fff"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Add
    </title>
    <path
      d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
    />
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
  </svg>
</button>
`;
